readthedocs/projects/tasks/search.py

```diff
@@ -40,112 +42,215 @@ def fileify(version_pk, commit, build, search_ranking, search_ignore):
     storage_path = version.project.get_storage_path(
-        type_='html', version_slug=version.slug, include_file=False
+        type_="html", version_slug=version.slug, include_file=False
     )
+    html_files_to_index = []
+    html_files_to_save = []
+    reverse_rankings = list(reversed(search_ranking.items()))
     for root, __, filenames in build_media_storage.walk(storage_path):
         for filename in filenames:
             # We don't care about non-HTML files
-            if not filename.endswith('.html'):
+            if not filename.endswith(".html"):
                 continue
 
             full_path = build_media_storage.join(root, filename)
 
             # Generate a relative path for storage similar to os.path.relpath
-            relpath = full_path.replace(storage_path, '', 1).lstrip('/')
+            relpath = full_path.replace(storage_path, "", 1).lstrip("/")
+
+            skip_search_index = False
+            if version.is_external:
+                # Never index files from external versions.
+                skip_search_index = True
+            else:
+                for pattern in search_ignore:
+                    if fnmatch(relpath, pattern):
+                        skip_search_index = True
+                        break
 
             page_rank = 0
-            # Last pattern to match takes precedence
-            # XXX: see if we can implement another type of precedence,
-            # like the longest pattern.
-            reverse_rankings = reversed(list(search_ranking.items()))
-            for pattern, rank in reverse_rankings:
-                if fnmatch(relpath, pattern):
-                    page_rank = rank
-                    break
-
-            ignore = False
-            for pattern in search_ignore:
-                if fnmatch(relpath, pattern):
-                    ignore = True
-                    break
-
-            # Create imported files from new build
-            HTMLFile.objects.create(
+            # If the file is ignored, we don't need to check for its ranking.
+            if not skip_search_index:
+                # Last pattern to match takes precedence
+                # XXX: see if we can implement another type of precedence,
+                # like the longest pattern.
+                for pattern, rank in reverse_rankings:
+                    if fnmatch(relpath, pattern):
+                        page_rank = rank
+                        break
+
+            html_file = HTMLFile(
                 project=version.project,
                 version=version,
                 path=relpath,
                 name=filename,
                 rank=page_rank,
-                commit=commit,
-                build=build,
-                ignore=ignore,
+                # TODO: We are setting this field since it's required,
+                # but it isn't used, and will be removed in the future
+                # together with other fields.
+                commit="unknown",
```

Comment:
Same idea that I mentioned before. Make it explicit.

```suggestion
                # TODO: We are setting the "commit" field since it's required,
                # but it isn't used, and will be removed in the future
                # together with other fields.
                commit="unknown",
```